### PR TITLE
fix: off-by-one: minimum positive result of getOutputPrice() is 1

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
[Formal Spec of Constant Product Market Maker](https://github.com/runtimeverification/verified-smart-contracts/blob/uniswap/uniswap/x-y-k.pdf) shows that `getOutputPrice()` should add one to the result of a complex division. I omitted the `+1`, which let `getOutputPrice()` return 0 rather than rounding up to one when the result of the division was quite small.

getOutputPrice

### CODE REVIEW

File: packages/zoe/src/contractSupport/bondingCurves.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ export const getOutputPrice = (
   const numerator = BigInt(outputValue) * BigInt(inputReserve) * BIG_10000;
   const denominator =
     (BigInt(outputReserve) - BigInt(outputValue)) * oneMinusFeeScaled;
-  return Nat(Number(numerator / denominator));
+  return Nat(Number(numerator / denominator)) + 1;
```
Comment (katelynsills) [inline packages/zoe/src/contractSupport/bondingCurves.js]: should we use natSafeMath here instead of adding with `+`?

**Review by katelynsills — CHANGES_REQUESTED** [review]
Looks good, but I think we should use the safeMath addition instead of `+` for consistency

File: packages/zoe/src/contractSupport/bondingCurves.js (hunk 1 above)
Comment (erights) [reply packages/zoe/src/contractSupport/bondingCurves.js]: We should do neither. We should do the addition within the BigInt world before converting back to the precision limited world.
```suggestion
  return Nat(Number(numerator / denominator + BigInt(1)));
```

File: packages/zoe/src/contractSupport/bondingCurves.js [hunk 2]
```diff
@@ -98,7 +99,7 @@ export const getOutputPrice = (
   const numerator = BigInt(outputValue) * BigInt(inputReserve) * BIG_10000;
   const denominator =
     (BigInt(outputReserve) - BigInt(outputValue)) * oneMinusFeeScaled;
-  return Nat(Number(numerator / denominator));
+  return Nat(Number(BIG_ONE + numerator / denominator));
```
Comment (katelynsills) [inline packages/zoe/src/contractSupport/bondingCurves.js]: A nitpick, but I think it is easier to read and understand with this order. It also matches the Uniswap implementation here: https://github.com/Uniswap/uniswap-v1/blob/c10c08d81d6114f694baa8bd32f555a40f6264da/contracts/uniswap_exchange.vy#L120

```suggestion
  return Nat(Number(numerator / denominator + BIG_ONE));
```

**Review by katelynsills — APPROVED** [review]
LGTM!

**Review by erights — APPROVED** [review]
LGTM

**Pull request merged**